Add support for named capture groups

ChrisPenner opened the pull request:
Attempting to refresh https://github.com/Daniel-Diaz/pcre-light/pull/9 .

Do you have a specific set of GHC's you want this to build for? I notice it's set to `7.6` and `7.8` ATM which is pretty old. 

Looking at the previous travis builds it's saying that it can't find GHC `7.6` or `7.8` so I suspect they've dropped support for testing them. I added `8.6.4` in this update, so we'll see if that one succeeds.

Let me know which versions you'd like to test on, we'll see if this succeeds.

ChrisPenner commented:
Looks like tests are passing now; so let me know which GHC's we want and I'll update the list.

I'll also add some tests for this new feature since it looks like there aren't any yet.

ChrisPenner commented:
Appears as though it wasn't set up to run unit tests on Travis; so I've added those to the cabal file, as well as adding tests for the capture names. I had to clean a bit of the oldest stuff up to get it to build, but changed as little as possible.

ChrisPenner commented on tests/Unit.hs:
```diff
@@ -3896,18 +3910,18 @@ tests = TestList
-         Nothing,
-         Nothing]
-    ,
+    -- testRegex "^[a-\\d]" []
```
This is the only test that fails; so far as I can tell this isn't a valid regex, so I'm unsure why this test is here. Do you have any ideas? Currently it's failing to compile the regex with the error:

```
Error in regex: invalid range in character class
```

Daniel-Diaz replied:
I didn't write this code, so I am not sure either. But since it doesn't seem valid, it's ok to remove it.

ChrisPenner commented on pcre-light.cabal:
```diff
@@ -17,7 +17,7 @@ license-file:    LICENSE
 author:          Don Stewart
 maintainer:      Daniel Díaz <dhelta.diaz@gmail.com>
-cabal-version: >= 1.2.0
+cabal-version: >= 1.8.0
```
Required to add a test-suite

Daniel-Diaz commented:
Thanks a lot! This is a great upgrade.

How about this set of versions? https://github.com/Daniel-Diaz/HaTeX/blob/master/.travis.yml#L2

ChrisPenner commented:
I removed the invalid test; switched the tested GHC versions, and bumped up the minor version in the cabal file.

With that it should be ready to go! Let me know if there's anything else you need 👍 

(P.S. If you're looking for an additional maintainer I have a few extra cycles to help out with that)

Daniel-Diaz commented:
Great! Thanks @ChrisPenner.

I merged the PR and I will release it to Hackage today.

And yes, an extra maintainer for pcre-light would be very good to have.

ChrisPenner commented:
@Daniel-Diaz Sounds good! Feel free to add me as a collaborator if you wish; I'll take a look at the other open PR when I have a chance 👍
